### PR TITLE
Refactor opencl_info

### PR DESCRIPTION
**Pull request by cbachhuber**
This pullrequest refactors parts of opencl information dumping: 

- Remove code duplication
- Clarify functionality of ternary operator
- Use cv's String instead of char * for strings
- Remove unused code

```
force_builders=Custom,Linux AVX2,Linux OpenCL
build_image:Custom=ubuntu:18.04
buildworker:Custom=linux-5
test_opencl:Custom=ON

build_image:Linux AVX2=ubuntu:18.04
buildworker:Linux AVX2=linux-3
test_opencl:Linux AVX2=ON
```

### CODE REVIEW

**Comment by asmorkalov** [comment]
@i-murzov Please take a look.

File: modules/core/include/opencv2/core/opencl/opencl_info.hpp [hunk 1]
```diff
@@ -94,15 +111,10 @@ static void dumpOpenCLInformation()
...
-        DUMP_CONFIG_PROPERTY("cv_ocl_current_platformName", device.getPlatform().name());
-#endif
-
-        const char* deviceTypeStr = (device.type() == Device::TYPE_CPU) ? "CPU" :
-            (device.type() == Device::TYPE_GPU ? device.hostUnifiedMemory() ? "iGPU" : "dGPU" : "unknown");
+        String deviceTypeStr = getDeviceTypeString(
+            device.type(), device.hostUnifiedMemory());
         DUMP_MESSAGE_STDOUT("    Type = " << deviceTypeStr);
-        DUMP_CONFIG_PROPERTY("cv_ocl_current_deviceType", deviceTypeStr);
+        DUMP_CONFIG_PROPERTY("cv_ocl_current_deviceType", deviceTypeStr.c_str());
```
Comment (GArik) [inline modules/core/include/opencv2/core/opencl/opencl_info.hpp]: unnecessary change

File: modules/core/include/opencv2/core/opencl/opencl_info.hpp [hunk 2]
```diff
@@ -94,15 +111,10 @@ static void dumpOpenCLInformation()
...
         DUMP_MESSAGE_STDOUT("Current OpenCL device: ");
 
-#if 0
-        DUMP_MESSAGE_STDOUT("    Platform = " << device.getPlatform().name());
-        DUMP_CONFIG_PROPERTY("cv_ocl_current_platformName", device.getPlatform().name());
-#endif
-
-        const char* deviceTypeStr = (device.type() == Device::TYPE_CPU) ? "CPU" :
-            (device.type() == Device::TYPE_GPU ? device.hostUnifiedMemory() ? "iGPU" : "dGPU" : "unknown");
+        String deviceTypeStr = getDeviceTypeString(
```
Comment (GArik) [inline modules/core/include/opencv2/core/opencl/opencl_info.hpp]: excessive line splitting

File: modules/core/include/opencv2/core/opencl/opencl_info.hpp [hunk 3]
```diff
@@ -80,12 +97,12 @@ static void dumpOpenCLInformation()
             for (int j = 0; j < platform->deviceNumber(); j++)
             {
                 platform->getDevice(current_device, j);
-                const char* deviceTypeStr = (current_device.type() == Device::TYPE_CPU) ? "CPU" :
-                    (current_device.type() == Device::TYPE_GPU ? current_device.hostUnifiedMemory() ? "iGPU" : "dGPU" : "unknown");
+                String deviceTypeStr = getDeviceTypeString(
+                    current_device.type(), current_device.hostUnifiedMemory());
```
Comment (GArik) [inline modules/core/include/opencv2/core/opencl/opencl_info.hpp]: excessive line splitting

File: modules/core/include/opencv2/core/opencl/opencl_info.hpp [hunk 4]
```diff
@@ -47,6 +47,23 @@ static std::string bytesToStringRepr(size_t value)
...
+  if (deviceType == cv::ocl::Device::TYPE_GPU) {
+    if (hostUsesUnifiedMemory) {
+      return "iGPU";
+    } else {
+      return "dGPU";
+    }
+  }
+
+  return "unkown";
+}
```
[the remark this answers is not in the document]
Comment (cbachhuber) [reply modules/core/include/opencv2/core/opencl/opencl_info.hpp]: Good catch, thanks!

File: modules/core/include/opencv2/core/opencl/opencl_info.hpp [hunk 5]
```diff
@@ -47,6 +47,23 @@ static std::string bytesToStringRepr(size_t value)
         s = s.substr(0, s.size() - 1);
     return s;
 }
+
+static String getDeviceTypeString(const int deviceType,
+                                  const bool hostUsesUnifiedMemory) {
```
[the remark this answers is not in the document]
Comment (cbachhuber) [reply modules/core/include/opencv2/core/opencl/opencl_info.hpp]: Good points, I agree

File: modules/core/include/opencv2/core/opencl/opencl_info.hpp (hunk 1 above)
Comment (cbachhuber) [reply modules/core/include/opencv2/core/opencl/opencl_info.hpp]: In this function, there is no cv::format as in line 103, right.

File: modules/core/include/opencv2/core/opencl/opencl_info.hpp (hunk 2 above)
Comment (cbachhuber) [reply modules/core/include/opencv2/core/opencl/opencl_info.hpp]: Resolved by shorter function signature

File: modules/core/include/opencv2/core/opencl/opencl_info.hpp (hunk 3 above)
Comment (cbachhuber) [reply modules/core/include/opencv2/core/opencl/opencl_info.hpp]: Resolved by shorter function signature

**Comment by asmorkalov** [comment]
LGTM :+1: @cbachhuber please squash commits.

File: modules/core/include/opencv2/core/opencl/opencl_info.hpp [hunk 6]
```diff
@@ -47,6 +47,22 @@ static std::string bytesToStringRepr(size_t value)
         s = s.substr(0, s.size() - 1);
     return s;
 }
+
+static String getDeviceTypeString(const cv::ocl::Device& device) {
```
Comment (GArik) [inline modules/core/include/opencv2/core/opencl/opencl_info.hpp]: Braces should go on a new line. Or you can drop them entirely when they are unnecessary. 
Otherwise, the code looks good to me.

Comment (cbachhuber) [reply modules/core/include/opencv2/core/opencl/opencl_info.hpp]: Again, correct. Actually, does opencv have an autoformat file (clang-format) that I can apply?

**Comment by cbachhuber** [comment]
Thank you for the reviews, squashed commits

File: modules/core/include/opencv2/core/opencl/opencl_info.hpp (hunk 6 above)
Comment (alalek) [reply modules/core/include/opencv2/core/opencl/opencl_info.hpp]: Some coding style information is available on Wiki: https://github.com/opencv/opencv/wiki/Coding_Style_Guide#code-layout

`clang-format` looks weird (introduces too many unnecessary changes - there is no way to specify "keep"/"do not care" option for some rules) - you may find PR with these experiments.

Comment (cbachhuber) [reply modules/core/include/opencv2/core/opencl/opencl_info.hpp]: Got it, thanks a lot for the info!
